pennylane/ops/op_math/controlled_ops.py

```diff
@@ -334,3 +340,894 @@ def compute_matrix():  # pylint: disable=arguments-differ
 
     def _controlled(self, wire):
         return CCZ(wires=wire + self.wires)
+
+
+class CRX(ControlledOp):
+    r"""The controlled-RX operator
+
+    .. math::
+
+        \begin{align}
+            CR_x(\phi) &=
+            \begin{bmatrix}
+            & 1 & 0 & 0 & 0 \\
+            & 0 & 1 & 0 & 0\\
+            & 0 & 0 & \cos(\phi/2) & -i\sin(\phi/2)\\
+            & 0 & 0 & -i\sin(\phi/2) & \cos(\phi/2)
+            \end{bmatrix}.
+        \end{align}
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+    * Gradient recipe: The controlled-RX operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://doi.org/10.1088/1367-2630/ac2cb3):
+
+      .. math::
+
+          \frac{d}{d\phi}f(CR_x(\phi)) = c_+ \left[f(CR_x(\phi+a)) - f(CR_x(\phi-a))\right] - c_- \left[f(CR_x(\phi+b)) - f(CR_x(\phi-b))\right]
+
+      where :math:`f` is an expectation value depending on :math:`CR_x(\phi)`, and
+
+      - :math:`a = \pi/2`
+      - :math:`b = 3\pi/2`
+      - :math:`c_{\pm} = (\sqrt{2} \pm 1)/{4\sqrt{2}}`
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wire the operation acts on
+        id (str or None): String representing the operation (optional)
+    """
+    num_wires = 2
+    """int: Number of wires that the operation acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    name = "CRX"
+    basis = "X"
```

Comment:
```suggestion
```

Basis should also be inherited from `SymbolicOp.basis`

https://github.com/PennyLaneAI/pennylane/blob/c71d85066909b08119539742c35b43529efb9478/pennylane/ops/op_math/symbolicop.py#L102